### PR TITLE
Add log messages to frames

### PR DESCRIPTION
**Pull request by SpotlightKid**
(implements #93)

Log messages can be added with the `-m/--message` option to the `stop` command or via the `edit` command to the current or completed frames.

This is still work in progress to get early feedback. This PR depends on #115.

Currently it is not possible to change `watson.current`, because the getter method always returns a copy of the dict. Thatswhy the `cli.stop` function has to change `watson._current` directly, which seems a bit unclean. Is there a specific reason why the `watson.current` getter function is implemented this way?

Documentation is missing and will be added in due time.

I'd also like to add support for displaying log messages to the `log` and `report` command but IMHO this should be done in a separate PR. Also, it's probably better to wait with it until #117 is merged.

### CODE REVIEW

**Comment by jmaupetit** [comment]
Looks good to me. Thank you for this! We will give some impulse to merge pending PRs.

**Comment by SpotlightKid** [comment]
Ready for review.

Like I said, support for log messages in `log` and `report` command to follow in another PR.

**Comment by k4nar** [comment]
LGTM :+1: 

That's an awesome chunk of work, congrats :) .

File: scripts/conv_frames.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+from __future__ import print_function
+
+import json
+import os
+import tempfile
+
+import click
+
+
+app_dir = click.get_app_dir('watson')
```
Comment (SpotlightKid) [inline scripts/conv_frames.py]: I added support for this in PR #115, from which this port is derived. But please note that I wrote this script only for the convenience of developers, switching between old and new versions of Watson. The `frames` module, as implemented in PR #115 is backwards-compatible, it  detects a frame file in the old format automatically and converts it while loading it.

**Comment by jmaupetit** [comment]
Let's merge this after #115

**Comment by JordanKnott** [comment]
Is this PR going to be merged? Comments are a feature I definitely would like (which is why I maintain a personal fork of this project). Is there anything I can do to help move things along?

**Comment by SpotlightKid** [comment]
The branch needs serious updating and it also depends on another PR, which needs updating too.

I'll try to have a look at it after Christmas.

**Comment by JoeCotellese** [comment]
@SpotlightKid @JordanKnott - has anyone picked up the work on this PR? I'd love to get this merged.

**Comment by SpotlightKid** [comment]
@JoeCotellese I'm sorry to say that I have personally stopped using Watson and therefor have not pursued development further. If the creators of Watson want take over the work from this PR, they are free to do so.

**Comment by jnsebgosselin** [comment]
@jmaupetit I can take this over if you agree.

**Comment by jmaupetit** [comment]
@SpotlightKid Thank you again for your amazing contributions to this project ❤️ 
@jnsebgosselin Yes, sure! But there is a lot of work on this as it's tied to #115 Don't hesitate to ask for help if required. Thanks!

**Comment by jnsebgosselin** [comment]
@jmaupetit Should I open a new PR to continue the work on this?

**Comment by jmaupetit** [comment]
Yes, why not!

**Comment by JoeCotellese** [comment]
thanks for picking this up @jnsebgosselin let me know if I can help you with testing anything.

**Comment by jmaupetit** [comment]
We are letting @jnsebgosselin taking part of its free time to work on this (if he still has some). We are not hurried.

**Comment by jnsebgosselin** [comment]
I'm planning to complete this during the holidays

**Comment by ThomasWaldmann** [comment]
It would be really cool to attach comments/messages to frames.

For me, this feature still missing is a total show stopper: I can't use watson without this feature as I would have to keep separate notes (which would be kind of silly).

Potential applications:
- what precisely was done in some specific timeframe
- sometimes "why did it take that long to solve, problems encountered"
- sometimes "why was it not solved"
- could also include some rate specification, e.g. if you bill different kinds of work at different rates
- anything else not possible with watson yet in a formalized way, but can be easily put into a comment

**Comment by varac** [comment]
> For me, this feature still missing is a total show stopper: I can't use watson without this feature as I would have to keep separate notes (which would be kind of silly).

Same for me - super important feature. Please implement it soon !

**Comment by jnsebgosselin** [comment]
Sorry for that. I have a working solution based on this PR that I've been using for a couple of months now, I just need to take the time to submit a PR. I'll try to do it tonight...

**Comment by SpotlightKid** [comment]
Closing this in favour of #252 and since i have no interest anymore in getting this merged.

**Pull request closed without merging**